@xen-orchestra/lite/src/libs/relative-time.ts

```diff
@@ -0,0 +1,31 @@
+const DIVISIONS = [
+  { amount: 60, name: "seconds" },
+  { amount: 60, name: "minutes" },
+  { amount: 24, name: "hours" },
+  { amount: 7, name: "days" },
+  { amount: 4.34524, name: "weeks" },
+  { amount: 12, name: "months" },
+  { amount: Number.POSITIVE_INFINITY, name: "years" },
+] as const;
+
+export const toRelativeTime = (
+  from: Date | number,
+  to: Date | number,
+  locale: string
+) => {
+  const formatter = new Intl.RelativeTimeFormat(locale, {
+    numeric: "auto",
+  });
+
+  let duration = (new Date(from).getTime() - new Date(to).getTime()) / 1000;
+
+  for (const division of DIVISIONS) {
+    if (Math.abs(duration) < division.amount) {
+      return formatter.format(Math.round(duration), division.name);
```

Comment:
1.49 hours would be rounded down to 1 hour and it feels a bit too much. Should we allow 1 decimal, maybe? I'm not sure it's a good idea because I think we rarely see "1.1 hours". It's more common to have "1 hour 6 minutes" but then it would get more complicated. Let me know what you think.
/cc @MathieuRA @Rajaa-BARHTAOUI